src/Test.UnitTests.Sarif/Core/SarifLogTests.cs

```diff
@@ -200,6 +204,152 @@
             newSarifLog.Runs[0].Results.Count.Should().Be(sarifLog.Runs[0].Results.Count);
         }
 
+        [Fact]
+        public async Task SarifLog_PostStream_WithInvalidParameters_ShouldThrowArgumentNullException()
+        {
+            var testCases = new[]
+            {
+                new
+                {
+                    Title = "Empty 'postUri' parameter",
+                    PostUri = string.Empty,
+                    Stream = (Stream)new MemoryStream(),
+                    HttpClient = new HttpClient()
+                },
+                new
+                {
+                    Title = "Null 'stream' parameter",
+                    PostUri = "https://github.com/microsoft/sarif-sdk",
+                    Stream = (Stream)null,
+                    HttpClient = new HttpClient()
+                },
+                new
+                {
+                    Title = "Null 'httpClient' parameter",
+                    PostUri = "https://github.com/microsoft/sarif-sdk",
+                    Stream = (Stream)new MemoryStream(),
+                    HttpClient = (HttpClient)null
+                },
+            };
+
+            var sb = new StringBuilder();
+
+            foreach (var testCase in testCases)
+            {
+                Exception exception = await Record.ExceptionAsync(async () =>
+                {
+                    await SarifLog.Post(testCase.PostUri,
+                                        testCase.Stream,
+                                        testCase.HttpClient);
+                });
+
+                if (exception == null || exception.GetType() != typeof(ArgumentNullException))
+                {
+                    sb.AppendLine($"The test '{testCase.Title}' was expecting 'false' value.");
+                }
+            }
+
+            sb.Length.Should().Be(0, sb.ToString());
+        }
+
+        [Fact]
+        public async Task SarifLog_PostFile_WithInvalidParameters_ShouldThrowException()
+        {
+            string postUri = string.Empty;
+            string filePath = string.Empty;
+            var fileSystem = new Mock<IFileSystem>();
+
+            Exception exception = await Record.ExceptionAsync(async () =>
+            {
+                await SarifLog.Post(postUri,
+                                    filePath,
+                                    fileSystem.Object,
+                                    httpClient: null);
+            });
+
+            exception.Should().BeOfType(typeof(ArgumentNullException));
+
+            filePath = "SomeFile.txt";
+            fileSystem
+                .Setup(f => f.FileExists(It.IsAny<string>()))
+                .Returns(false);
+
+            exception = await Record.ExceptionAsync(async () =>
```

Comment:
>ExceptionAsync [](http://example.com/codeflow?start=37&length=14)

Can't we simplify this using Assert.ThrowsAsync? Or am I missing something?

 await Assert.ThrowsAsync<ArgumentNullException>(() => MethodThatThrows());

Reply:
Assert.ThrowsAsync is from MSTest. Record.Exception/ExceptionAsync is from xunit.

if we are going to use xunit everywhere, then, we should keep using record.exception.

if we dont care about using two different test patterns, then, i can move to assert.throws

:)

Reply:
just replaced to asser.throw style since we are normally using it.